tests/ui/coherence/coherence-free-regions.rs

```diff
@@ -0,0 +1,12 @@
+//@ check-pass
+
+trait Trait<'a> {}
+
+impl Trait<'static> for u8 {}
+
+impl Trait<'_> for (u8,) {}
+impl<T> Trait<'_> for (T,) where T: for<'a> Trait<'a> {}
```

Comment:
this test lacks a comment explaining its purpose. Probably easiest to remove it as I am already adding it in https://github.com/rust-lang/rust/pull/119820/files#diff-0ecb72e4c24504e07286feda773be9f35d0af1ff8a10d4820078c25f6b3dc0d0